pkg/controller/pxcrestore/restore.go

```diff
@@ -101,11 +107,30 @@ func (r *ReconcilePerconaXtraDBClusterRestore) createJob(job *batchv1.Job) error
 	for {
 		time.Sleep(time.Second * 1)
 
+		pods := &corev1.PodList{}
+		ls := map[string]string{
+			"job-name": job.Name,
+		}
+		err = r.client.List(context.TODO(), pods, &client.ListOptions{
+			Namespace:     job.Namespace,
+			LabelSelector: labels.SelectorFromSet(ls),
+		})
```

Comment:
one job can rerun pods multiple times - https://github.com/percona/percona-xtradb-cluster-operator/blob/main/pkg/pxc/backup/restore.go#L219 (this value should be configurable in future)
but according to this logic, only the first failed retry will fail the whole job.

Reply:
@delgod fixed in 24baf3e0f5e71623a4805ff5848b9081fd45a3fb